lib/table.js

```diff
@@ -380,7 +380,9 @@ Table.prototype.destroy = function (hashKey, rangeKey, options, callback) {
 
   let params = {
     TableName: self.tableName(),
-    Key: self.serializer.buildKey(hashKey, rangeKey, self.schema)
+    Key: self.serializer.buildKey(hashKey, rangeKey, self.schema),
+    ExpressionAttributeNames: { '#hashKey': self.schema.hashKey },
```

Comment:
I think that for the purposes of checking whether an item exists in the database, checking for the presence of the `hash key` is sufficient, even if the record also has a `range key`.  I closed this PR because this is already possible using the `expected` parameter like `{ hashKey: { Exists: true } }`.  You could also specify the `range key` exists in this constraint.  I just think it would be a redundant check.